presto-product-tests/bin/run_on_docker.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash -ex
+
+# http://stackoverflow.com/questions/3572030/bash-script-absolute-path-with-osx
+function absolutepath() {
+    [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"
+}
+
+ENVIRONMENT=$1
+
+if [[ "$ENVIRONMENT" != "singlenode" && "$ENVIRONMENT" != "distributed" ]]; then
+   echo "Usage: run_on_docker.sh <singlenode|distributed> <product test args>"
+   exit 1
+fi
+
+shift 1
+
+SCRIPT_DIR=$(dirname $(absolutepath "$0"))
+PRODUCT_TESTS_ROOT="${SCRIPT_DIR}/.."
+PROJECT_ROOT="${PRODUCT_TESTS_ROOT}/.."
+DOCKER_COMPOSE_LOCATION="${PRODUCT_TESTS_ROOT}/conf/docker/${ENVIRONMENT}/docker-compose.yml"
+DOCKER_LOGS_LOCATION="/tmp/presto_docker_logs"
+
+function retry() {
+  END=$(($(date +%s) + 600))
+
+  while (( $(date +%s) < $END )); do
+    set +e
+    "$@"
+    EXIT_CODE=$?
+    set -e
+
+    if [[ ${EXIT_CODE} == 0 ]]; then
+      break
+    fi
+    sleep 5
+  done
+
+  return ${EXIT_CODE}
+}
+
+function check_hadoop() {
+  HADOOP_MASTER_CONTAINER=$(docker ps --format '{{.Names}}' | grep hadoop-master)
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl status hive-server2 | grep -i running
+}
+
+function stop_unnessery_hadoop_services() {
+  HADOOP_MASTER_CONTAINER=$(docker ps --format '{{.Names}}' | grep hadoop-master)
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl status
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop mapreduce-historyserver
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop yarn-resourcemanager
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop yarn-nodemanager
+  docker exec ${HADOOP_MASTER_CONTAINER} supervisorctl stop zookeeper
+}
+
+function check_presto() {
+  DOCKER_NETWORK=$(docker network ls | grep ${ENVIRONMENT} | cut  -f 1 -d ' ')
+  docker run \
+    --rm -it \
+    --net ${DOCKER_NETWORK} \
+    -v "${PROJECT_ROOT}/presto-cli/target/:/cli" \
+    teradatalabs/centos6-java8-oracle \
+    java -jar /cli/presto-cli-${PRESTO_VERSION}-executable.jar --server presto-master:8080 \
+    --execute "SHOW CATALOGS"  | grep -i hive
+}
+
+function run_product_tests() {
+    DOCKER_NETWORK=$(docker network ls | grep ${ENVIRONMENT} | cut  -f 1 -d ' ')
```

Comment:
Docker compose creates the network and removes one as part of the `down` cycle. So it should be called strictly after the docker compose is called. We have these functions on top, so using the constants from bottom within them could be confusing. But from the other side you are totally right, this is copy-paste, what is definitely bad. As a compromise i will extract the network command into the separate functions, say `determine_docker_network`.